### ISSUE TITLE
Asset Browser: GM crashes/unstable when duplicating an asset in Feb24 Beta

### ISSUE THREAD

**Issue by SamTHodson** [YoYoGames/GameMaker-Bugs#4493]
### Description

When duplicating an asset in the latest Feb24 Beta, GM will either go unstable if this is done via the context menu or crash if done via the short (CTRL+D). 

### Unstable Error
```
[09:39:51:208(9001)] Exception in YoYoStudio.GUI.Gadgets.ContextMenu+<>c__DisplayClass140_0,Void <InitialiseEntry>b__0() :    System.ArgumentNullException: Value cannot be null. (Parameter 'key')
   at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
   at System.Collections.Generic.Dictionary`2.ContainsKey(TKey key)
   at YoYoStudio.Resources.ProjectFileWatcher.IgnoreResource(ResourceBase _res) in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Core\CoreResources\CoreResources\Management\ProjectFileWatcher.cs:line 240
   at YoYoStudio.Plugins.AssetBrowser.AssetTreeContextMenu.MenuDuplicate() in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Plugins\AssetBrowser\FastTree\AssetTreeContextMenu.cs:line 1129
   at WeakDelegate(Object , Object[] )
   at YoYoStudio.Core.Utils.WeakDelegate`1.Invoke(Object[] _args) in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Core\Utils\WeakDelegate.cs:line 184
[09:39:51:208(9001)] Fatal Error! 'IDE.OnWeakDelegateException - Value cannot be null. (Parameter 'key')'
[09:39:51:209(9001)]    at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
   at System.Collections.Generic.Dictionary`2.ContainsKey(TKey key)
   at YoYoStudio.Resources.ProjectFileWatcher.IgnoreResource(ResourceBase _res) in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Core\CoreResources\CoreResources\Management\ProjectFileWatcher.cs:line 240
   at YoYoStudio.Plugins.AssetBrowser.AssetTreeContextMenu.MenuDuplicate() in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Plugins\AssetBrowser\FastTree\AssetTreeContextMenu.cs:line 1129
   at WeakDelegate(Object , Object[] )
   at YoYoStudio.Core.Utils.WeakDelegate`1.Invoke(Object[] _args) in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Core\Utils\WeakDelegate.cs:line 184
[09:39:51:209(9001)] ####################################
[09:39:51:209(9001)] Break in FatalErrorWarning (IDE:4542)
[09:39:51:209(9001)] ####################################
``` 

### Crash Error
```
[09:41:11:709(9001)] IDE Exception: System.ArgumentNullException: Value cannot be null. (Parameter 'key')
   at System.Collections.Generic.Dictionary`2.FindValue(TKey key)
   at System.Collections.Generic.Dictionary`2.ContainsKey(TKey key)
   at YoYoStudio.Resources.ProjectFileWatcher.IgnoreResource(ResourceBase _res) in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Core\CoreResources\CoreResources\Management\ProjectFileWatcher.cs:line 240
   at YoYoStudio.Plugins.AssetBrowser.AssetTreeContextMenu.MenuDuplicate() in D:\a\YoYoStudio\YoYoStudio\YoYoStudio\IDE\Plugins\AssetBrowser\FastTree\AssetTreeContextMenu.cs:line 1129
   at YoYoStudio.KeyboardShortcuts.Shortcut.Execute()
   at YoYoStudio.KeyboardShortcuts.KeyboardShortcutManager.TryExecuteShortcut(KeyPress keyPress, IShortcutContext context, Action preExecute)
   at YoYoStudio.KeyboardShortcuts.KeyboardShortcutManager.<>c__DisplayClass52_0.<TryExecuteShortcut>g__TryExecute|1(Nullable`1 keyPress, IShortcutContext context)
   at YoYoStudio.KeyboardShortcuts.KeyboardShortcutManager.<WalkGadgetStack>g__WalkUpGadgetStack|54_1(<>c__DisplayClass54_0& )
   at YoYoStudio.KeyboardShortcuts.KeyboardShortcutManager.WalkGadgetStack(GUIBase gadgetInFocus, ExecuteDelegate executeAction, Boolean exitOnNoKeyPress)
   at YoYoStudio.KeyboardShortcuts.KeyboardShortcutManager.TryExecuteShortcut(GUIBase gadgetInFocus)
   at YoYoStudio.GUI.KeyboardManager.ProcessEvents(Desktop _desktop, GUIBase _gadgetInFocus, Boolean _dragging, Boolean _blocking)
   at YoYoStudio.IDE.Run()
   at YoYoStudio.IDE.Execute(String[] _args, Int32 _initialWidth, Int32 _initialHeight)
[09:41:11:709(9001)] --------------------------------------------------------------------------------
[09:41:11:709(9001)] ####################################
[09:41:11:710(9001)] Break in FatalErrorWarning (IDE:4542)
[09:41:11:710(9001)] ####################################
``` 
Tested using Beta IDE v2024.200.0.472  Runtime v2024.200.0.488

[Contact Us Package](https://drive.google.com/file/d/1QJfQNkyez3ZENZU3-I-p0Q_2h2CnZ3d0/view?usp=sharing)

### Steps To Reproduce

1. Start GameMaker Beta and create/open a project
2. Create any asset
3. Right-click the asset and select "Duplicate"
4. Observe the unstable message
5. Use CTRL+D to duplicate the asset
6. Observe the IDE crashes

Repro: 100%

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.200 (Betas)

### Which operating system(s) are you seeing the problem on?

Windows 11, Windows 10

### Are you running GameMaker from inside your Steam library?

No

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by stuckie** [comment]
Fixed as part of 7361ac026903c4aaa780121355c6cb06eb5dfb12

**Comment by SamTHodson** [comment]
Verified in Beta IDE v2024.200.0.475  Runtime v2024.200.0.490
